### DISCUSSION TITLE
wire:loading.class issue causing flood of syncInput requests

### DISCUSSION THREAD

**Discussion by benjaminboruff** [livewire/livewire#3937] [1. Help] [answered]
Is something odd happening when using the wire:loading directive with the class modifier? In this specific case, when used in a `@forelse` loop associated with a transaction search query in (this case specifically) a "Dashboard" livewire component, the `wire:loading.class="whatever"` directive on table rows in the blade view causes non-stop `syncInput` requests to fire in the browser when additional input is typed after the initial search finds no results.

For example (using my test [repo's](https://github.com/benjaminboruff/wireloading) simplified code based on Caleb's screencast `Building Data Tables -> Basic Search`), quickly type in a search string (in this case, yourapp/dashboard) that should produce `No transactions found...`. If you typed quickly enough, you should see the round-trip in the networking tab. Then type one or more additional letters ... then boom, non-stop `syncInput` requests begin to fire.

To stop it you must backspace over the search string until some records are found. I've tried other `wire:loading` directives, but none but the `class` modifier seems to trigger this behavior.

I think Caleb also ran into this issue in the `Basic Search` video about time marker *9:48* when he says `"Oh, I have some loopty-dupe thing going on"`. If you look in the bottom left corner of his screen you will see the `"Waiting for surge.test..." ` messages flashing in the browser. That's when I think he hit the problem. I think refreshing, then not triggering it again, may have stopped the issue during the remaining time of the video.

I am using both Chrome and Firefox (latest) on linux, running laravel 8.62.0, php 8.0.11 in Homestead 11.4.0, with livewire 2.6.7, and alpine 3.x.x (loaded in layouts.base)..

So setup, copy the example .env file and add your DB stuff. Then migrate with seed (this gives you 250 transactions from the factory). Then go to your site's dashboard route, and quickly type in a bunch of gibberish in the search input. You should get `No transactions found...` in the table.

Then type some additional characters and look at your browser's networking tab in the dev tools. You should see a barrage of `syncInput` requests.

Anyone else experience this? Am I doing something wrong? I'm new to livewire ... (and a sponsor for Caleb!).
Any info would be great!

Here are the files in question, the component/controller  and the view:

```php
<?php

namespace App\Http\Livewire;

use App\Models\Transaction;
use Livewire\Component;
use Livewire\WithPagination;

class Dashboard extends Component
{
//    use WithPagination;
    public $search = '';

    public function render()
    {
        return view('livewire.dashboard', [
            'transactions' => Transaction::where('title', 'like', '%'.$this->search.'%')->get(),
//            'transactions' => Transaction::where('title', 'like', '%'.$this->search.'%')->paginate(10),
        ]);
    }
}
```

  
```blade
<div>
    <h1>Dashboard</h1>

    <main>
        <div>
            <div>
                <input wire:model="search" type="text" placeholder="Search transactions...">
            </div>
            <table class="">
                <thead>
                <tr>
                    <th>Title</th>
                    <th>Amount</th>
                    <th>Status</th>
                    <th>Date</th>
                </tr>
                </thead>

                <tbody>
                @forelse($transactions as $transaction)
                    <tr wire:loading.class="">
                        <td> {{ $transaction->title }} </td>
                        <td> {{ $transaction->amount }} </td>
                        <td> {{ $transaction->status }} </td>
                        <td> {{ $transaction->date }} </td>
                    </tr>
                @empty
                    <tr>
                        <td colspan="4">
                            <span>
                                No transactions found...
                            </span>
                        </td>

                    </tr>
                @endforelse

                </tbody>
            </table>

            <div>
                {{--                {{ $transactions->links() }}--}}
            </div>

        </div>
    </main>
</div>
```

**Comment by benjaminboruff** [comment, marked as answer]
It was a  dom diffing issue. I added a `wire:key` to the table body rows, and now all is good. sheesh. 

```blade
<tbody>
  @forelse($transactions as $transaction)
      <tr wire:loading.class="" wire:key="{{ $loop->index }}">
          <td> {{ $transaction->title }} </td>
          <td> {{ $transaction->amount }} </td>
          <td> {{ $transaction->status }} </td>
          <td> {{ $transaction->date }} </td>
      </tr>
  @empty
      <tr>
          <td colspan="4">
              <span>
                  No transactions found...
              </span>
          </td>
  
      </tr>
  @endforelse
</tbody>
```

**Comment by adriaanzon** [comment]
I was also having issues with a flood of requests from version 2.5.2. On version 2.5.1 and earlier I didn't have this problem.

I was also able to fix it using wire:key. I now specify a different key for each variable that could cause attributes on the element to change, like this:

```blade
<button
    class="flex w-full text-left px-4 py-0.5 transition-colors duration-150 @if ($active) bg-sky-100 @elseif ($count === 0) cursor-not-allowed opacity-50 @else hover:bg-cool-gray-100 @endif"
    x-on:click="$dispatch('input', {{ json_encode($active ? [] : [$option]) }})"
    wire:loading.class="cursor-wait"
    @if ($count === 0) disabled @else wire:loading.attr="disabled" @endif
    wire:key="{{ class_basename($filter) }}-option:{{ $option }}-active:{{ json_encode($active) }}-count:{{ $count }}"
>
    <div class="flex-1">{{ $filter->humanReadableValue($option) }}</div>
    <span class="text-cool-gray-400">{{ $count }}</span>
</button>
```
